src/engine/strat_engine/backstore/devices.rs

```diff
@@ -264,6 +264,54 @@ pub struct DeviceInfo {
     pub size: Bytes,
 }
 
+/// Devices that have all been identified as Stratis devices.
+#[derive(Debug)]
+pub struct StratisDevices {
+    inner: HashMap<PoolUuid, HashMap<DevUuid, DeviceInfo>>,
+}
+
+impl StratisDevices {
+    // Given a pool UUID partition the devices into two divisions;
+    // those that belong to the pool and those that do not.
+    fn partition(
+        mut self,
+        uuid: PoolUuid,
+    ) -> (Option<HashMap<DevUuid, DeviceInfo>>, StratisDevices) {
+        let this_pool = self.inner.remove(&uuid);
+        (this_pool, StratisDevices { inner: self.inner })
+    }
+
+    // Return an error message on the assumption that these devices have
+    // been identified as belonging to another pool.
+    fn error_on_not_empty(&self) -> StratisResult<()> {
+        let error_string = self
+            .inner
+            .iter()
+            .map(|(pool_uuid, devs)| {
+                format!(
+                    "devices ({}) appear to belong to Stratis pool with UUID {}",
+                    devs.iter()
+                        .map(|(_, info)| info.devnode.display().to_string())
+                        .collect::<Vec<_>>()
+                        .join(", "),
+                    pool_uuid
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("; ");
```

Comment:
I think it may make sense to remove this here and invoke it on line 308.

For example:

```
let error_strings = ... .collect::<Vec<_>>();

...

let error_message = format!(..., error_strings.join("; "));
```

I don't think testing for an empty string is as clear as testing for an empty `Vec`.

Reply:
Done!